modules/RTC/JitsiLocalTrack.js

```diff
@@ -144,6 +158,34 @@ function JitsiLocalTrack(
 JitsiLocalTrack.prototype = Object.create(JitsiTrack.prototype);
 JitsiLocalTrack.prototype.constructor = JitsiLocalTrack;
 
+JitsiLocalTrack.prototype._addPeerConnection = function(tpc) {
+    if (this._isAttachedToPC(tpc)) {
+        logger.error(`${tpc} has been associated with ${this} already !`);
+    } else {
+        this.peerConnections.add(tpc);
+    }
+};
+
+JitsiLocalTrack.prototype._removePeerConnection = function(tpc) {
+    if (this._isAttachedToPC(tpc)) {
+        this.peerConnections.delete(tpc);
+    } else {
+        logger.error(`${tpc} is not associated with ${this}`);
+    }
+};
+
+/**
+ * Checks whether or not this instance is attached to given
+ * <tt>TraceablePeerConnection</tt>. See
+ * {@link TraceablePeerConnection.attachTrack} for more info.
+ * @param {TraceablePeerConnection.attachTrack} tpc
+ * @return {boolean} <tt>true</tt> if this tracks is currently attached to given
+ * peer connection or <tt>false</tt> otherwise.
+ */
+JitsiLocalTrack.prototype._isAttachedToPC = function(tpc) {
```

Comment:
none of these things are really public, the problem is that all the fields in all of the classes are like that currently. It will take too much time to fix everything now